src/stravalib/client.py

```diff
@@ -189,17 +195,51 @@ def exchange_code_for_token(
             The developer client secret
         code : str
             The temporary authorization code
+        return_athlete : bool (default = False)
+            Whether to return a SummaryAthlete object (or not)
+            This parameter is currently undocumented and could change
+            at any time.
 
         Returns
         -------
-        dict
-            Dictionary containing the access_token, refresh_token and
+        AccessInfo
+            TypedDictionary containing the access_token, refresh_token and
             expires_at (number of seconds since Epoch when the provided access
             token will expire)
+        tuple
+            Contains:
+             - the `AccessInfo` typed dict containing token values
+             - a `SummaryAthlete` object representing the authenticated user.
+
+        Notes
+        -----
+        Strava by default returns `SummaryAthlete` information during
+        this exchange. However this return is currently undocumented
+        and could change at any time.
         """
-        return self.protocol.exchange_code_for_token(
-            client_id=client_id, client_secret=client_secret, code=code
-        )
+        if return_athlete:
```

Comment:
This is much cleaner. However, mypy is not confused about protocol returning an "iterable". I'm not sure how to make it happy and could use any help @jsamoocha @enadeau . thank you!  I tried to "cast" it to a tuple but that didn't work.